### PR TITLE
Enum type basic create/insert/select support

### PR DESCRIPTION
**Pull request by iceTTTT**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #6613

## What this PR does / why we need it:

doc https://github.com/iceTTTT/docs/blob/enum/design/enum.md

### CODE REVIEW

File: pkg/util/export/etl/tae.go [hunk 1]
```diff
@@ -398,7 +398,7 @@ func GetVectorArrayLen(ctx context.Context, vec *vector.Vector) (int, error) {
 		cols := vector.MustFixedCol[types.Datetime](vec)
 		return len(cols), nil
 	default:
-		return 0, moerr.NewInternalError(ctx, "the value type %d is not support now", *vec.GetType())
+		return 0, moerr.NewInternalError(ctx, "the value type with oid %d is not support now", vec.GetType().Oid)
```
Comment (xzxiong) [inline pkg/util/export/etl/tae.go:401]: lgtm

**Review by xzxiong — APPROVED** [review]

File: pkg/vm/engine/tae/catalog/schema.go [hunk 2]
```diff
@@ -254,6 +249,19 @@ func (s *Schema) ReadFrom(r io.Reader) (n int64, err error) {
 			return
 		}
 		n += sn
+		var length uint64
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/catalog/schema.go:252]: As we talked offline, there is no need to change here. Need a modification in types.EncodeType

File: pkg/vm/engine/tae/containers/batch.go [hunk 3]
```diff
@@ -198,7 +198,12 @@ func (bat *Batch) WriteTo(w io.Writer) (n int64, err error) {
 	for i, vec := range bat.Vecs {
 		buffer.Append([]byte(bat.Attrs[i]))
 		vt := vec.GetType()
-		buffer.Append(types.EncodeType(&vt))
+		var data []byte
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/containers/batch.go:201]: Same. Not to use types.Encode and types.Decode

File: pkg/vm/engine/tae/containers/batch.go [hunk 4]
```diff
@@ -254,7 +259,9 @@ func (bat *Batch) ReadFrom(r io.Reader) (n int64, err error) {
 		bat.Attrs[i] = string(buf)
```
Comment (XuPeng-SH) [inline pkg/vm/engine/tae/containers/batch.go:259]: Same

**Review by aressu1985 — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Comment by iceTTTT** [comment]
image
go benchmark for edited encode/decodeType without enumstring.
image
Previous encode/decodeType benchmark.
Got about 3-4 times degression.

**Pull request closed without merging**